tomcat/tomcat-jta/src/main/java/org/jboss/narayana/tomcat/jta/TransactionalDataSourceFactory.java

```diff
@@ -119,50 +67,41 @@ public Object getObjectInstance(Object obj, Name name, Context context, Hashtabl
         }
 
         final Properties properties = new Properties();
-        final Properties generic_pool_properties = new Properties();
-        final Properties base_pool_properties = new Properties();
-        final Properties poolable_connection_properties = new Properties();
-        final Properties abandoned_properties = new Properties();
         Enumeration<RefAddr> iter = ref.getAll();
 
         while (iter.hasMoreElements()) {
             RefAddr ra = iter.nextElement();
             String type = ra.getType();
             String content = ra.getContent().toString();
-            if (GENERIC_POOLING_PROPERTIES.contains(type)) {
-                generic_pool_properties.setProperty(type, content);
-            } else if (BASE_POOLING_PROPERTIES.contains(type)) {
-                base_pool_properties.setProperty(type, content);
-            } else if (POOLABLE_CONNECTION_PROPERTIES.contains(type)) {
-                poolable_connection_properties.setProperty("_" + type, content);
-            } else if (ABANDONED_PROPERTIES.contains(type)) {
-                abandoned_properties.setProperty(type, content);
-            } else if (PROP_USERNAME.equals(type)) {
-                properties.setProperty(type, content);
-            } else if (PROP_PASSWORD.equals(type)) {
-                properties.setProperty(type, content);
-            }
+            properties.setProperty(type, content);
         }
 
         final TransactionManager transactionManager = (TransactionManager) getReferenceObject(ref, context, PROP_TRANSACTION_MANAGER);
         final XADataSource xaDataSource = (XADataSource) getReferenceObject(ref, context, PROP_XA_DATASOURCE);
 
         if (transactionManager != null && xaDataSource != null) {
-            final DataSourceXAConnectionFactory xaConnectionFactory =
-                    new DataSourceXAConnectionFactory(transactionManager, xaDataSource);
-            final PoolableConnectionFactory poolableConnectionFactory =
-                    new PoolableConnectionFactory(xaConnectionFactory, null);
-            final GenericObjectPoolConfig config = new GenericObjectPoolConfig();
-            final AbandonedConfig abandonedConfig = new AbandonedConfig();
+            String initialSize = properties.getProperty("initialSize");
```

Comment:
I was wondering why initialSize is different to other properties? Perhaps you can add a comment into the code to explain?

Reply:
There is a trick to fix [DBCP-215](https://issues.apache.org/jira/browse/DBCP-215) so we have to remove the "initialSize" that the BaiscDataSourceFactory.createDataSource(properties) will not create the connections in the pool. And it will create with the BaiscManagedDataSource later.

I will add these comments in the code.